### ISSUE TITLE
logging.disable(logging.INFO) conflicts with Shiny's logging and also prevents it from launching browser on start ("--launch-browser")

### ISSUE THREAD

**Issue by Dene33** [rstudio/py-shiny#569]
I am currently using the `logging` package in my code, and Shiny also depends on it for message logging. So how can I disable logging only in my code? `logging.disable(logging.INFO)` not only disables Shiny logs but also prevents it from launching the browser on start ("--launch-browser" doesn't work).

The answer is not to use the root logger, create a new one instead. However, blocking "--launch-browser" while disabling the root logger is weird.

**Comment by jcheng5** [comment]
Yeah, it's very surprising, but we use log interception to know when uvicorn's child process (that actually runs the Shiny app) is ready to handle requests. It's weird and fragile but the other approaches all had worse downsides. That said, we could detect that the root logger is disabled and use a fallback strategy (like spawning a thread that repeatedly tries to establish a socket connection).

**Comment by bradwbradw** [comment]
This also happens if simply setting `--log-level=warning`

here is my full command:

```
  uv run shiny run main.py
  --host $APP_HOST
  --port $APP_PORT
  --reload
  --reload-dir $SCRIPT_DIR/app
  --dev-mode
  --log-level=info
  --launch-browser
```

`--log-level=info` launches the browser fine, but `--log-level=warning` does not

-----
*AI take:*
Why it happens:

In Shiny, browser launch is tied to a log handler that listens for the startup message Application startup complete.
That handler is attached at INFO level in the Shiny internals.
When you run with log level warning, that INFO startup event is not emitted to the handler, so launch-browser never fires.
I confirmed this in:

venv/lib/python3.11/site-packages/shiny/_main.py
venv/lib/python3.11/site-packages/shiny/_launchbrowser.py
venv/lib/python3.11/site-packages/shiny/_autoreload.py

-----

Perhaps an easy fix is just change the level of the log ready message to `critical`, so it gets captured no matter what level is set?
